chore: update mkt lifecycle spec (#1366)
vegaprotocol/specs: protocol/0043-MKTL-market_lifecycle.md

### Cancelled

A market becomes Cancelled when a Market Proposal is successful and conditions are not met to transition the Market to the Active state during the Pending period, 

> and the trading terminated data source input rings, see [data sourcing](./0045-DSRC-data_sourcing.md).

Reviewer: Is there term "rings" here one that is clearly and unambiguously understood across the project? It confused me at first but by the end of reviewing the PR I understood the meaning but am not sure if it's a well understood term or not.

Reviewer: After discussion and failure to come up with a good and succinct way to describe lets approve. I was unsure, and if it's unclear someone can raise an issue.

> When a market transitions to a cancelled state all orders should be cancelled and collateral returned to respective parties general account for the relevant asset, all LP commitments should be cancelled and their bond returned to the general account for the relevant asset and any insurance pool balance should be transferred into the network treasury account for that asset. 

Reviewer: Should we also mention that there must also be no open positions in a cancelled state (even though as currently designed none can be created)?

Author: I'll add that.

Author: Changed my mind - it'll sound confusing to say that there can't be positions - given that we talk about returning collateral etc.

Reviewer: I meant here that there must be no open positions once we reach the cancelled state, not before transitioning, i.e. if there are open positions they must be closed from the perspective of both the core's position tracking and sending updates (e.g. event bus msgs) to close them. I think we should still include something to this effect. Have a suggested change.

Reviewer: ```suggestion
When a market transitions to a cancelled state all orders should be cancelled and collateral returned to respective parties general account for the relevant asset, all LP commitments should be cancelled and their bond returned to the general account for the relevant asset and any insurance pool balance should be transferred into the network treasury account for that asset. 

Once "cancelled" there must be no open positions tracked by the protocol for the market and any open positions must have been closed including returning all margin and other related collateral if necessary and also notifying downstream event consumers that the positions are closed. Specific position related actions may be unnecessary if the cancelled state is being entered from a state in which there cannot possibly have been any open positions.
```

* the Instrument hits expiry; or
* governance votes (by approving a proposal to close the market) not to create the market after all.
A market becomes Cancelled when a Market Proposal is successful and conditions are not met to transition the Market to the Active state during the Pending period, 
and the trading terminated data source input rings, see [data sourcing](./0045-DSRC-data_sourcing.md).
When a market transitions to a cancelled state all orders should be cancelled and collateral returned to respective parties general account for the relevant asset, all LP commitments should be cancelled and their bond returned to the general account for the relevant asset and any insurance pool balance should be transferred into the network treasury account for that asset. 
All data sources that are only referenced by this market should be unregistered. 

> In future, it's expected that we will implement the functionality described in the white paper by which a proposer can revoke their proposal during the voting period by voting against it. In this case the status would also become Cancelled.

Reviewer: I would like to implement this feature I think, though I am ok with not stating so here if that's preferable.

Author: I agree. 

I've removed it for now since with the growth of QA and them taking more ownership of specs (both good) it's harder to keep overall understanding of which bits of specs are to be implemented and which ones are really in *now* or to be done.

### Trading Terminated

> A market will terminate trading if the trading terminated trigger data source rings.

Reviewer: This is only true for the cash settled future product and I don't think that product specific functionality should be elevated to system level here.

```suggestion
A market may terminate trading if the product is one that expires or if the market is otherwise configured to have a finite lifetime.
```

> In the case of futures, termination occurs at some point prior to the settlement of the product. Markets in this state accept no trading, but retain the positions and margin balances that were in place after processing the trading terminated trigger (which may itself generate MTM cashflows, though for futures it doesn't). 

Reviewer: ```suggestion
In the case of futures, termination occurs at some point prior to the settlement of the product and is triggered by the trading terminated data source configured on the market. Markets in this state accept no trading, but retain the positions and margin balances that were in place after processing the trading terminated trigger (which may itself generate MTM cashflows, though for futures it doesn't). 
```

Any insurance pool balance is transferred to the network treasury account for the asset. 
The market state is `settled`. 

### Market never leaves opening auction, trading terminated trigger rings, market cancelled (<a name="0043-MKTL-003" href="#0043-MKTL-003">0043-MKTL-003</a>)

> 1. A market is proposed, approved by governace process and enterst the opening auction (Pending state).

Reviewer 2: small typo on "enters"

Author: 👍

1. Trading terminated data source rings before the market leaves the opening auction (so market never left Pending state so far).
1. All orders should be cancelled and collateral returned to respective parties general account for the relevant asset.
1. All LP commitments should be cancelled and their bond returned to the general account for the relevant asset.

> 1. Any insurance pool balance should be transferred into the network treasury account for that asset. 

Reviewer 2: not sure this applies. If market never leaved initial Pending state, there is no way we can have anything in the insurance pool (no positions, no trading, no distress, etc)

Author: In the future it will be possible to fund insurance pool with transfers. 

But yes, at this point you won't be able to test this.
